materialize-sqlserver/sqlgen.go

```diff
@@ -0,0 +1,288 @@
+package main
+
+import (
+	"strings"
+	"time"
+	"fmt"
+	"text/template"
+
+	"slices"
+	sql "github.com/estuary/connectors/materialize-sql"
+)
+
+var sqlServerDialect = func(stringType string, collation string) sql.Dialect {
+	var textType = fmt.Sprintf("%s(MAX) COLLATE %s", stringType, collation)
+	var textPKType = fmt.Sprintf("%s(900) COLLATE %s", stringType, collation)
+
+	// nvarchar and varchar fields require the value to be a string instead of a
+	// bytearray to handle unicode properly, so we convert to str after converting
+	// to json.RawMessage
+	var jsonConverter = sql.WithElementConverter(sql.Compose(sql.StdByteArrayToStr, sql.JsonBytesConverter))
```

Comment:
@williamhbaker I see a few differences here between these implementations:

1. The `jsonConverter` of bigquery is not marshalling non-raw-jsons to json, it is erroring in those cases
2. In case of a `nil` value of `[]byte` or `json.RawMessage`, the bigquery implementation is returning `string(nil)` which is an empty string `""`, but in this case, I want the value to be `nil`, as they are different. I want the database to insert `nil` into the column, not an empty string

Reply:
For where we are using the `jsonConverter`, it should only ever get types of `[]byte` for Flow objects and arrays, or `json.RawMessage` for the document. It will get `nil` if the values are absent or `null`.

Good point about the bigquery implementation not handling the `nil` case correctly - it should be returning `nil` instead of an empty string there, so I'd call that a deficiency.

I'm anticipating some broader refactorings around how we are converting values in SQL materializations, especially now that we are making it more dynamic with some of the intricacies surfaced for the MySQL and this SQL Server materialization, so I think this is something we can resolve later.

Reply:
Oh I'm actually wrong: We are using `jsonConverter` for SQL server the handle `multiple` types as well, which could be anything. Anyway, this LGTM for the moment.